File: gestionrh-backend/src/main/java/com/gestionrh/gestionrhbackend/services/admin/AdminService.java
package com.gestionrh.gestionrhbackend.services.admin;

import com.gestionrh.gestionrhbackend.dtos.CompanyDto;

import java.io.IOException;
import java.util.List;

public interface AdminService {
    //CompanyDto postCompany(CompanyDto companyDto) throws IOException;
    boolean postCompany(CompanyDto companyDto) throws IOException;

    List<CompanyDto> getAllCompanies();

    void deleteCompany(Long id);

    CompanyDto getCompanyById(Long id);

    boolean updateCompany(Long companyId, CompanyDto companyDto) throws IOException;
}

File: src/app/modules/admin/admin-services/admin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {Observable} from "rxjs";
import {StorageService} from "../../../auth-services/storage-service/storage.service";

const BASIC_URL = ["http://localhost:8080/"]

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor(private http: HttpClient) { }

  postCompany(companyDto: any):Observable<any>{
    return this.http.post<[]>(BASIC_URL + "api/admin/company",companyDto,{
      headers: this.createAuthorizationHeader()
    })
  }

  getAllCompanies():Observable<any>{
    return this.http.get(BASIC_URL + "api/admin/companies",{
      headers: this.createAuthorizationHeader()
    })
  }

  deletCompany(id:number):Observable<any>{
    return this.http.delete(BASIC_URL + "api/admin/company/" + id,{
      headers: this.createAuthorizationHeader()
    });
  }

  getCompanyById(id: number):Observable<any>{
    return this.http.get(BASIC_URL + "api/admin/company/" + id,{
      headers: this.createAuthorizationHeader()
    });
  }

  updateCompany(companyId: number, companyDto: any): Observable<any> {
    return this.http.put(BASIC_URL + "api/admin/company/" + companyId, companyDto,{
      headers: this.createAuthorizationHeader()
    });
  }
  createAuthorizationHeader():HttpHeaders{
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      'Authorization','Bearer ' + StorageService.getToken()
    );
  }

}
